docs/csharp/language-reference/statements/exception-handling-statements.md

```diff
@@ -0,0 +1,141 @@
+
+  :::code language="csharp" source="snippets/exception-handling-statements/Program.cs" id="ThrowExpressionConditional":::
+
+- [the null-coalescing operator](../operators/null-coalescing-operator.md). The following example uses a `throw` expression to throw an <xref:System.ArgumentNullException> when the string to assign to a property is `null`:
+
+  :::code language="csharp" source="snippets/exception-handling-statements/ExampleClass.cs" id="ThrowExpressionCoalescing":::
+
+- an expression-bodied [lambda](../operators/lambda-expressions.md) or method. The following example uses a `throw` expression to throw an <xref:System.InvalidCastException> to indicate that a conversion to a <xref:System.DateTime> value is not supported:
+
+  :::code language="csharp" source="snippets/exception-handling-statements/Program.cs" id="ThrowExpressionExpressionBody":::
+
+## The `try` statement
+
+You can use the `try` statement in any of the following forms: [`try-catch`](#the-try-catch-statement) - to handle exceptions that might occur during execution of the code inside a `try` block, [`try-finally`](#the-try-finally-statement) - to specify the code that is executed when control leaves the `try` block, and [`try-catch-finally`](#the-try-catch-finally-statement) - as a combination of the preceding two forms.
+
+### The `try-catch` statement
+
+Use the `try-catch` statement to handle exceptions that might occur during execution of a code block. Place the code where an exception might occur inside a `try` block. Use a *catch clause* to specify the base type of exceptions you want to handle in the corresponding `catch` block:
+
+:::code language="csharp" source="snippets/exception-handling-statements/Program.cs" id="TryCatch":::
+
+You can use several catch clauses:
+
+:::code language="csharp" source="snippets/exception-handling-statements/Program.cs" id="TryMultipleCatch":::
+
+When an exception occurs, catch clauses are examined in the specified order, from top to bottom. At maximum, only one `catch` block is executed for any thrown exception. As the preceding example also shows, you can omit declaration of an exception variable and specify only the exception type in a catch clause. A catch clause without any specified exception type matches any exception and, if present, must be the last catch clause.
+
+If you want to re-throw a caught exception, use the [`throw` statement](#the-throw-statement), as the following example shows:
+
+:::code language="csharp" source="snippets/exception-handling-statements/Program.cs" id="RethrowInCatch":::
+
+> [!NOTE]
+> `throw;` preserves the original stack trace of the exception, which is stored in the <xref:System.Exception.StackTrace?displayProperty=nameWithType> property. Opposite to that, `throw e;` updates the <xref:System.Exception.StackTrace> property of `e`.
+
+#### A `when` exception filter
```

Comment:
We need to add text here to explain that exception filters modify the rules on the order of `catch` clauses.  Namely:

1. Multiple `catch` clauses can specify the same `Exception` type, if `when` clauses distinguish them.
2. The `catch Exception` clause doesn't need to be last *if* an exception filter is present.